### ISSUE TITLE
не враховує останьої години 23:59

### ISSUE THREAD

**Issue by sp0ngeboobs** [tsdaemon/ha-lviv-poweroff#9]
Скрипт не враховує останьої години. 
Наприклад в даному випадку що на скріншоті, скрипт заніс в календар період лише 22:00-23:00
так розумію через те що останя година вказана з хвилинами (23:59) які скрипт не враховує. 
<img width="400" alt="Screenshot_2" src="https://github.com/user-attachments/assets/71f646ac-55e7-4215-940c-1b15f2089659" />

**Comment by sp0ngeboobs** [comment]
у графіках відключень появились хвилини. Тепер половина графіків будуть невірно підтягуватись в HA

**Comment by Crickus** [comment]
Тут з'явилось таке ноу-хау як півгодинні інтервали.
![изображение](https://github.com/user-attachments/assets/db357d69-85cd-4575-9cfb-5b3ddb6e7ae3)
Зрозуміло, що в сенсорах вони повністю ігноруються)

**Comment by stenvix** [comment]
Привіт, зробив зміни по парсеру, була змога перевірити тільки на 5.1 групі, можливо комусь буде корисно) 
https://github.com/stenvix/ha-lviv-poweroff

**Comment by sp0ngeboobs** [comment]
@stenvix Супер, дякууу-єє-мо. Попередньо працює. Як будуть баги то відпишу.

**Comment by sp0ngeboobs** [comment]
@stenvix а UPDATE_INTERVAL = 600 в тебе робоча функція? бо я надалі користуюсь автоматизацією перезавантаження інтеграції.

**Comment by stenvix** [comment]
@sp0ngeboobs ні, не працює, але постараюсь до того глянути, можливо вдасться виправити

**Comment by dabakovich** [comment]
Всім привіт!

Виглядає так, що ~я~ AI agent пофіксив цю проблему з автооновленням. В оригінальній лібі закладено інтервал оновлення 600 секунд (10 хвилин), але цей функціонал був зламаний, та не працював.

Можете тестувати якщо є охота: https://github.com/dabakovich/ha-lviv-poweroff

Крім того, там додані відсутні групи, а також дані скрапаються напряму з Львівобленерго (тепер додає хвилини в графік, якщо початок кінець відключення припадає на половину години).

Фікси додав тільки вчора, тому щось не працюватиме, то створюйте PR.

**Comment by gresuk** [comment]
> Можете тестувати якщо є охота: https://github.com/dabakovich/ha-lviv-poweroff

Привіт. графіки відключення то появляються в календарі то пропадають, незнаю чи це проблема нового оновлення os 17.0 яке вчора прилетіло. 

<img width="569" height="569" alt="Image" src="https://github.com/user-attachments/assets/4dc5913f-9984-4dc0-9bd5-7ca63c93b4b1" />

**Comment by dabakovich** [comment]
@gresuk я теж це помітив. Але також помітив, що графіки реально мінялись на сайті. А оскільки оновлення тепер коректно відбувається кожних 10 хвилин, то інтеграція записує всі ці зміни. Але суть в тому, що перед самим відключенням інтеграція буде мати найбільш актуальну і свіжу інформацію. В моєму випадку надсилається сповіщення перед відключенням світла, то тепер воно дійсно приходить тоді, коли є найбільш актуальна інформація про реальне відключення.

Архів всіх змін графіків можна глянути на сторінці "Архів графіків" в ЛОЕ: https://loe.lviv.ua/page/arhiv_grafikiv
